src/group_theory/group_action/fixing_subgroup.lean

```diff
@@ -0,0 +1,198 @@
+{ carrier := { ϕ : M | ∀ x : s, ϕ • (x : α) = x },
+  one_mem' := λ _, one_smul _ _,
+  mul_mem' := λ x y hx hy z, by rw [mul_smul, hy z, hx z], }
+
+lemma mem_fixing_submonoid_iff {s : set α} {m : M} :
+  m ∈ fixing_submonoid M s ↔ ∀ (y : α) (hy : y ∈ s), m • y = y :=
+⟨λ hg y hy, hg ⟨y, hy⟩, λ h ⟨y, hy⟩, h y hy⟩
+
+variable (α)
+
+/-- The Galois connection between fixing submonoids and fixed points of a monoid action -/
+theorem fixing_submonoid_fixed_points_gc : galois_connection
+  (order_dual.to_dual ∘ (fixing_submonoid M))
+  ((λ P : submonoid M, (fixed_points P α)) ∘ order_dual.of_dual) :=
+λ s P, ⟨λ h s hs p, h p.2 ⟨s, hs⟩, λ h p hp s, h s.2 ⟨p, hp⟩⟩
+
+lemma fixing_submonoid_antitone : antitone (λ (s : set α), fixing_submonoid M s) :=
+galois_connection.monotone_l (fixing_submonoid_fixed_points_gc M α)
+
+lemma fixed_points_antitone :
+  antitone (λ (P : submonoid M), fixed_points P α) :=
+(fixing_submonoid_fixed_points_gc M α).monotone_u.dual_left
+
+/-- Fixing submonoid of union is intersection -/
+lemma fixing_submonoid_union {s t : set α} :
+  fixing_submonoid M (s ∪ t) = (fixing_submonoid M s) ⊓ (fixing_submonoid M t) :=
+galois_connection.l_sup (fixing_submonoid_fixed_points_gc M α)
+
+/-- Fixing submonoid of Union is intersection -/
+lemma fixing_submonoid_Union {ι : Sort*} {s : ι → set α} :
+  fixing_submonoid M (⋃ i, s i) = ⨅ i, fixing_submonoid M (s i) :=
+galois_connection.l_supr (fixing_submonoid_fixed_points_gc M α)
+
+/-- Fixed points of sup of submonoids is intersection -/
+lemma fixed_points_submonoid_sup {P Q : submonoid M} :
+  fixed_points (P ⊔ Q : submonoid M) α = fixed_points P α ∩ fixed_points Q α :=
+galois_connection.u_inf (fixing_submonoid_fixed_points_gc M α)
+
+/-- Fixed points of supr of submonoids is intersection -/
+lemma fixed_points_submonoid_supr {ι : Sort*} {P : ι → submonoid M} :
+  fixed_points (supr P : submonoid M) α = ⋂ i, fixed_points (P i) α :=
+galois_connection.u_infi (fixing_submonoid_fixed_points_gc M α)
+
+end monoid
+
+section group
+
+open mul_action
+
+variables (M : Type*) {α : Type*} [group M] [mul_action M α]
+
+/-- The subgroup fixing a set under a `mul_action`. -/
+@[to_additive /-" The additive subgroup fixing a set under an `add_action`. "-/]
+def fixing_subgroup (s : set α) : subgroup M :=
+{ inv_mem' := λ _ hx z, by rw [inv_smul_eq_iff, hx z],
+  ..fixing_submonoid M s, }
+
+lemma mem_fixing_subgroup_iff {s : set α} {m : M} :
+  m ∈ fixing_subgroup M s ↔ ∀ (y : α) (hy : y ∈ s), m • y = y :=
+⟨λ hg y hy, hg ⟨y, hy⟩, λ h ⟨y, hy⟩, h y hy⟩
+
+variable (α)
+
+/-- The Galois connection between fixing subgroups and fixed points of a group action -/
+lemma fixing_subgroup_fixed_points_gc : galois_connection
+  (order_dual.to_dual ∘ (fixing_subgroup M))
+  ((λ P : subgroup M, (fixed_points P α)) ∘ order_dual.of_dual) :=
+λ s P, ⟨λ h s hs p, h p.2 ⟨s, hs⟩, λ h p hp s, h s.2 ⟨p, hp⟩⟩
+
+lemma fixing_subgroup_antitone : antitone (λ (s : set α), fixing_subgroup M s) :=
```

Comment:
```suggestion
lemma fixing_subgroup_antitone : antitone (fixing_subgroup M : set α → subgroup M) :=
```
To avoid the binder

Reply:
@AntoineChambert-Loir this is still not fixed.